src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt

```diff
@@ -0,0 +1,123 @@
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.managedindexmetadata.StepMetaData
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.step.Step
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.index.IndexRollupAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.index.IndexRollupRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.start.StartRollupAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.start.StartRollupRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.model.ISMRollup
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.support.WriteRequest
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.index.engine.VersionConflictEngineException
+import java.lang.Exception
+
+class AttemptCreateRollupJobStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    val ismRollup: ISMRollup,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step(name, managedIndexMetaData) {
+
+    private val logger = LogManager.getLogger(javaClass)
+    private var stepStatus = StepStatus.STARTING
+    private var info: Map<String, Any>? = null
+    private var rollupId: String? = null
+    private var previousRunRollupId: String? = null
+    private var hasPreviousRollupAttemptFailed: Boolean? = null
+
+    override fun isIdempotent() = true
+
+    override suspend fun execute(): Step {
+        previousRunRollupId = managedIndexMetaData.actionMetaData?.actionProperties?.rollupId
+        hasPreviousRollupAttemptFailed = managedIndexMetaData.actionMetaData?.actionProperties?.hasRollupFailed
+
+        // Creating a rollup job
+        val rollup = ismRollup.toRollup(indexName)
+        rollupId = rollup.id
+        logger.info("Attempting to create a rollup job $rollupId for index $indexName")
+
+        val indexRollupRequest = IndexRollupRequest(rollup, WriteRequest.RefreshPolicy.IMMEDIATE)
+
+        try {
+            val response = withContext(Dispatchers.IO) { client.execute(IndexRollupAction.INSTANCE, indexRollupRequest).actionGet() }
+            logger.info("Received status ${response.status.status} on trying to create rollup job $rollupId")
+
+            stepStatus = StepStatus.COMPLETED
+            info = mapOf("message" to getSuccessMessage(rollupId!!, indexName))
+        } catch (e: VersionConflictEngineException) {
+            val message = getFailedJobExistsMessage(rollupId!!, indexName)
+            logger.info(message)
+            if (rollupId == previousRunRollupId && hasPreviousRollupAttemptFailed != null && hasPreviousRollupAttemptFailed!!) {
```

Comment:
`hasPreviousRollupAttemptFailed != null && hasPreviousRollupAttemptFailed!!` can be changed to `hasPreviousRollupAttemptFailed == true`

Reply:
What did we decide these mean again?

rollupId == previousRunRollupId is true but other is false
and
rollupId == previousRunRollupId is false but other is true

Reply:
👍🏾